src/http.rs

```diff
@@ -123,6 +124,27 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut data_buf: Option<Vec<u8>> = Some(Vec::new());
+            let body = body.map_data(move |data| {
+                if let Some(buf) = &mut data_buf {
+                    buf.extend_from_slice(&data);
+                }
+
+                if data_buf.is_some() {
+                    if let Some(buf) = &mut data_buf {
+                        if buf.len() >= 20 {
+                            emit!(http_client::HttpBodyDataReceived {
+                                buf: buf
```

Comment:
Note that Rust has a shorthand for fields that match a local variable:
```suggestion
                                buf
```